services/sidecar/rollup/derivation/batch_builder.go

```diff
@@ -96,8 +96,15 @@ func (b *batchBuilder) Advance() {
 func (b *batchBuilder) getBatch(l1Head types.BlockID) ([]byte, error) {
 	// Force-build batch if necessary (timeout exceeded).
 	force := b.timeout != 0 && l1Head.GetNumber() >= b.timeout
-	log.Info("Trying to get batch", "force?", force)
-	batch, err := b.encoder.GetBatch(force)
+	log.Info("Trying to get batch", "curr_l1#", l1Head.GetNumber(), "timeout_l1#", b.timeout, "force?", force)
+	batch, err := b.encoder.Flush(force)
+	if force {
+		// If it's too late to sequence, the batch should just be dropped entirely.
+		timeoutExceeded := l1Head.GetNumber() >= b.timeout+b.cfg.GetSubSafetyMargin()
+		if timeoutExceeded {
+			return nil, fmt.Errorf("batch timeout exceeded: %w", err)
```

Comment:
change `timeoutExceed` to batch too old? it is a bit confusing since there's a normal timeout

Reply:
hmm maybe will consider later; following `op-node` for now (took some of these checks from the channel mgr there)